python/utils/OpaqueArguments.h

```diff
@@ -155,22 +155,22 @@ inline py::args simplifiedValidateInputArguments(py::args &args) {
         throw std::runtime_error("Cannot pass ndarray with shape != (N,).");
 
       arg = args[i].attr("tolist")();
+    } else if (py::isinstance<py::str>(arg)) {
+      arg = cudaq::pauli_word(py::cast<std::string>(arg));
```

Comment:
If you do it this way you may be able to remove the casting behavior from kernel_decorator

Reply:
> If you do it this way you may be able to remove the casting behavior from kernel_decorator

Hmm, removing that code seems to cause errors in `python/tests/kernel/test_kernel_features.py::test_pauli_word_input`:
```
RuntimeError: error: Can not handle conversion of python type <class 'str'> to MLIR type.
```